### ISSUE TITLE
内存泄露问题

### ISSUE THREAD

**Issue by LonelyTaker** [RVC-Boss/GPT-SoVITS#1218]
`_batch_audio_fragment = (self.vits_model.decode(
                        all_pred_semantic, _batch_phones, refer_audio_spec
                    ).detach()[0, 0, :])`

目前定位到这条语句存在明显的内存泄露现象。想请问下是什么原因导致？

**Comment by ChasonJiang** [comment]
> `_batch_audio_fragment = (self.vits_model.decode( all_pred_semantic, _batch_phones, refer_audio_spec ).detach()[0, 0, :])`
> 
> 目前定位到这条语句存在明显的内存泄露现象。想请问下是什么原因导致？

如果你是在正常推理过程中看到显存增长，那么这是正常现象。因为这是由transformer 的kv cache机制决定的。

**Comment by ChasonJiang** [comment]
@LonelyTaker 正常推理之后没有释放显存吗？

**Comment by LonelyTaker** [comment]
> @LonelyTaker 正常推理之后没有释放显存吗？

不是显存，内存未被释放

**Comment by LonelyTaker** [comment]
> @LonelyTaker 正常推理之后没有释放显存吗？

随着推理请求的增多，占用内存会不断增加，即使释放模型，也无法释放掉占用的内存

**Comment by ChasonJiang** [comment]
> 随着推理请求的增多，占用内存会不断增加，即使释放模型，也无法释放掉占用的内存

了解了

**Comment by ysujiang** [comment]
> > @LonelyTaker 正常推理之后没有释放显存吗？
> 
> 随着推理请求的增多，占用内存会不断增加，即使释放模型，也无法释放掉占用的内存

请问这个问题有解决吗

**Comment by LonelyTaker** [comment]
> > > @LonelyTaker 正常推理之后没有释放显存吗？
> > 
> > 
> > 随着推理请求的增多，占用内存会不断增加，即使释放模型，也无法释放掉占用的内存
> 
> 请问这个问题有解决吗

没
